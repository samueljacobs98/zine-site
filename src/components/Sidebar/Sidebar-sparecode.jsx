import React, { useState, useRef, useEffect } from "react";
import "./Sidebar.scss";

export default function SidebarNav() {
  const [isSidebar, initSidebar] = useState(true);

  const openSidebar = (isSidebar) => {
    // console.log(isSidebar)
    return initSidebar(!isSidebar);
  };

  const node = useRef();

  const trackSidebar = (e) => {
    if (node.current.contains(e.target)) {
      // inside scope click
      return;
    }
    // outside scope click
    initSidebar(false);
  };

  useEffect(() => {
    console.log("yo");
    document.addEventListener("mousedown", trackSidebar);
    return () => {
      document.removeEventListener("mousedown", trackSidebar);
    };
  }, []);

  return (
    <>
      <header className="header">
        <nav className="navbar">
          <h2>React Animated Sidebar Example</h2>
          <span
            ref={node}
            className="hamburger"
            onClick={() => {
              openSidebar(isSidebar);
            }}
          >
            <svg
              xmlns="http://www.w3.org/2000/svg"
              className="h-5 w-5"
              viewBox="0 0 20 20"
              fill="currentColor"
            >
              {isSidebar ? (
                <path
                  fillRule="evenodd"
                  d="M4.293 4.293a1 1 0 011.414 0L10 8.586l4.293-4.293a1 1 0 111.414 1.414L11.414 10l4.293 4.293a1 1 0 01-1.414 1.414L10 11.414l-4.293 4.293a1 1 0 01-1.414-1.414L8.586 10 4.293 5.707a1 1 0 010-1.414z"
                  clipRule="evenodd"
                />
              ) : (
                <path
                  fillRule="evenodd"
                  d="M3 5a1 1 0 011-1h12a1 1 0 110 2H4a1 1 0 01-1-1zM3 10a1 1 0 011-1h12a1 1 0 110 2H4a1 1 0 01-1-1zM3 15a1 1 0 011-1h12a1 1 0 110 2H4a1 1 0 01-1-1z"
                  clipRule="evenodd"
                />
              )}
            </svg>
          </span>
        </nav>

        <div
          ref={node}
          className="sidebar-block"
          style={{ left: isSidebar ? "0" : "-260px" }}
        >
          <a href="#">Home</a>
          <a href="#">About</a>
          <a href="#">Products</a>
          <a href="#">Portfolio</a>
          <a href="#">Profile</a>
          <a href="#">Sign out</a>
        </div>
      </header>
    </>
  );
}
