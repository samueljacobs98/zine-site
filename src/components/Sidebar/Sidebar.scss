.sidebar-block-open {
  animation: in 1s;
  position: fixed;
  left: 0px;
  top: 0px;
  height: 100%;
  width: 265px;
  background: black;
  transition: all 0.35s ease-out;
  border: solid 2px black;
  display: flex;
  flex-direction: column;
  align-items: center;
  padding: 30px 0;

  .sidebar-profile {
    display: flex;
    flex-direction: column;
    align-items: center;

    .sidebar-picture {
      border-radius: 50%;
      width: 180px;
      height: 180px;
      object-fit: cover;
    }
  }
  .sidebar-links {
    display: flex;
    align-items: center;
    flex-direction: column;

    .sidebar-link {
      text-decoration: none;
      color: white;
      padding: 1rem 1.5rem;
      border-bottom: 1px solid rgba(0, 0, 0, 0.1);
    }

    .sidebar-link:hover {
      cursor: pointer;
    }
  }

  .sidebar-button {
    border: none;
    background: none;
    justify-self: center;
    align-self: center;
    margin-top: auto;
  }

  .sidebar-button:hover {
    cursor: pointer;
    color: red;
  }
}

@keyframes in {
  from {
    left: -265px;
  }
  to {
    left: 0px;
  }
}
