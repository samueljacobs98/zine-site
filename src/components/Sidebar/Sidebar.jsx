import "./Sidebar.scss";
import sampleprofilepic from "../../assets/images/SamplePhoto.jpg";

const Sidebar = ({ showSidebar, username }) => {
  const animationStyle = showSidebar ? "open" : "close";
  return (
    <div className={` sidebar-block-${animationStyle}`}>
      <div className="sidebar-profile">
        <img className="sidebar-picture" src={sampleprofilepic} alt="" />
        <p className="sidebar-name">@{username}</p>
      </div>
      <div className="sidebar-links">
        <a className="sidebar-link" href="/">
          Home
        </a>
        <a className="sidebar-link" href="/">
          Followers
        </a>
        <a className="sidebar-link" href="/">
          Following
        </a>
        <a className="sidebar-link" href="/">
          Trending
        </a>
      </div>

      <button className="sidebar-button">LOG OUT</button>
    </div>
  );
};

export default Sidebar;
