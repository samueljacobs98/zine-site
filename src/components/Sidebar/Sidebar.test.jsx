import { customRender } from "../../utils/testUtils";
import Sidebar from "./Sidebar.jsx";

it("should match NavBar component snapshot", () => {
  const { container } = customRender(<Sidebar />);

  expect(container).toMatchSnapshot();
});
