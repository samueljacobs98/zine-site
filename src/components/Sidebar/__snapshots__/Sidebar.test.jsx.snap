// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match NavBar component snapshot 1`] = `
<div>
  <div
    class=" sidebar-block-close"
  >
    <div
      class="sidebar-profile"
    >
      <img
        alt=""
        class="sidebar-picture"
        src="SamplePhoto.jpg"
      />
      <p
        class="sidebar-name"
      >
        @
      </p>
    </div>
    <div
      class="sidebar-links"
    >
      <a
        class="sidebar-link"
        href="/"
      >
        Home
      </a>
      <a
        class="sidebar-link"
        href="/"
      >
        Followers
      </a>
      <a
        class="sidebar-link"
        href="/"
      >
        Following
      </a>
      <a
        class="sidebar-link"
        href="/"
      >
        Trending
      </a>
    </div>
    <button
      class="sidebar-button"
    >
      LOG OUT
    </button>
  </div>
</div>
`;
